src/opencl_diskcryptor_aes_fmt_plug.c

```diff
@@ -0,0 +1,416 @@
+/*
+ * This software is Copyright (c) 2018, Ivan Freed <ivan.freed at protonmail.com>,
+ * Copyright (c) 2012-2013 Lukas Odzioba, Copyright (c) 2014 JimF, Copyright
+ * (c) 2014 magnum, and it is hereby released to the general public under the
+ * following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Based on opencl_pbkdf2_hmac_sha512_fmt_plug.c file.
+ */
+
+#ifdef HAVE_OPENCL
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_opencl_diskcryptor_aes;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_opencl_diskcryptor_aes);
+#else
+
+#include <stdint.h>
+#include <string.h>
+
+#include "misc.h"
+#include "arch.h"
+#include "common.h"
+#include "formats.h"
+#include "options.h"
+#include "unicode.h"
+#include "opencl_common.h"
+#include "diskcryptor_common.h"
+#include "pbkdf2_hmac_common.h"
+
+#define FORMAT_NAME             "DiskCryptor AES XTS (only)"
+#define FORMAT_LABEL            "diskcryptor-aes-opencl"
+#define ALGORITHM_NAME          "PBKDF2-SHA512 AES OpenCL"
+#define BINARY_SIZE             0
+#define BINARY_ALIGN            MEM_ALIGN_WORD
+#define SALT_SIZE               sizeof(struct custom_salt)
+#define SALT_ALIGN              sizeof(uint32_t)
+#define PLAINTEXT_LENGTH        110
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      1
+#define KERNEL_NAME             "pbkdf2_sha512_kernel"
+#define SPLIT_KERNEL_NAME       "pbkdf2_sha512_loop"
+#define FINAL_KERNEL_NAME       "diskcryptor_final"
+
+#define HASH_LOOPS              250
+#define ITERATIONS              10000
```

Comment:
I just realized the `ITERATIONS` macro should be same as number of PBKDF2 iterations, I think 1000 here?

Reply:
That's for correct auto-tuning.